### PR TITLE
#164198198 Users can bookmark articles

### PR DESCRIPTION
**Pull request by fejiroofficial**
**What does this PR do?**
This is PR for building a feature that allows users to bookmark articles for a later read.

**Description of Task to be completed?**
- Write unit test(s)
- Write code and migration scripts
- Write Controller functions and validators to handle feature

**How should this be manually tested**
- Clone this repo and cd into it
- run `npm install` to install dependencies.
- test using mocha with `npm test`. 

**What are the relevant pivotal tracker stories**
[#164198198](https://www.pivotaltracker.com/n/projects/2313263/stories/164198198)

**Screenshots**
N/A

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (aanchirinah) [inline .vscode/launch.json]: #### Critical
- Explain why this file exists?

File: controllers/articlesController.js [hunk 2]
```diff
@@ -156,4 +159,84 @@ export default class ArticlesController {
...
   }
+
+  /**
+ * @description users can bookmark articles for reading later.
+ * @static
+ * @param  {object} req - The request object
+ * @param  {object} res - The response object
+ * @return {object} - It returns the request response object
+ */
+  static async bookmark(req, res) {
```
Comment (aanchirinah) [inline controllers/articlesController.js]: #### Nitpick
- Find a better name than `bookmark`, you could make it `createBookmark` or `bookmarkArticle` etc.

File: controllers/articlesController.js [hunk 3]
```diff
@@ -156,4 +159,84 @@ export default class ArticlesController {
...
+  }
+
+  /**
+* @description users can get all bookmarked articles.
+* @static
+* @param  {object} req - The request object
+* @param  {object} res - The response object
+* @return {object} - It returns the request response object
+*/
+  static async getBookmarkedArticles(req, res) {
```
Comment (aanchirinah) [inline controllers/articlesController.js]: #### Comment
- Good method name given to this one , be consistent 😄

File: routes/articles.js [hunk 4]
```diff
@@ -218,4 +218,10 @@ articles.post(
   ArticleLikeController.dislike
 );
```
Comment (aanchirinah) [inline routes/articles.js]: #### Critical
- Please add a Swagger Doc for this route.

File: controllers/articlesController.js (hunk 2 above)
Comment (fejiroofficial) [reply controllers/articlesController.js]: done

File: .vscode/launch.json (hunk 1 above)
Comment (fejiroofficial) [reply .vscode/launch.json]: It is coming from the debug tool. I will remove it. I thought it was a local file

File: controllers/articlesController.js (hunk 3 above)
Comment (fejiroofficial) [reply controllers/articlesController.js]: thank you

File: routes/articles.js (hunk 4 above)
Comment (fejiroofficial) [reply routes/articles.js]: done

File: controllers/articlesController.js [hunk 5]
```diff
@@ -156,4 +159,84 @@ export default class ArticlesController {
...
+ * @param  {object} res - The response object
+ * @return {object} - It returns the request response object
+ */
+  static async bookmark(req, res) {
+    const { slug } = req.params;
+    const userId = req.user.id;
+    let articleId;
+
+    try {
+      const user = await User.findOne({
```
Comment (Andraquin) [inline controllers/articlesController.js]: @fejiroofficial Starting from line 176 to 189, kindly leverage on an existing middleware for authenticating users and write a separate middleware for article validation.

Comment (fejiroofficial) [reply controllers/articlesController.js]: done

**Comment by aanchirinah** [comment]
@fejiroofficial , Please resolve conflicts.

**Comment by aanchirinah** [comment]
@fejiroofficial , Your build has failed.

**Pull request merged**